src/utils/filterResults.ts

```diff
@@ -37,58 +38,74 @@ function getOwnerData (property: UnprocessedResultsFromCRM) {
+        const desiredPropertyGroups = searchParams.propertyGroups
+        const maxResultsForPropertyTypes = searchParams.propertyTypesMaxResults
+        const maxResultsForPropertyGroups = searchParams.propertyGroupsMaxResults
+        const desiredManaged = searchParams.managed
+        const maxNumNeighbours = searchParams.neighboursSearchMaxRecords
+
+        const matchTallies: MatchTallies = {
+            neighbour: 0,
+            propertyType: 0,
+            propertyGroup: 0
+        }
+
+        const isPropertyTypeFilterInUse = desiredPropertyTypes.includes('Filter Disabled')
+        const isPropertyGroupFilterInUse = desiredPropertyGroups.includes('Filter Disabled')
+
+        unsortedPropertyResults.forEach((property: UnprocessedResultsFromCRM) => {
+            if (!property.Latitude || !property.Longitude) {
+                return
+            }
+            const isUnderNeighbourLimit = matchTallies.neighbour < maxNumNeighbours
+            const isUnderPropertyTypeLimit = matchTallies.propertyType < maxResultsForPropertyTypes
+            const isUnderPropertyGroupLimit = matchTallies.propertyGroup < maxResultsForPropertyGroups
+            let canAddBasedOnMaxResults = isUnderNeighbourLimit || isUnderPropertyTypeLimit || isUnderPropertyGroupLimit
+
+            if (filterInUse === 'SalesEvidenceFilter') {
+                // N.B. the Sales Evidence Filter doesn't have the ability to search for multiple properties hence only passing in the single search param object.
+                const allRecordsForSalesEvidenceFilter = searchParams.allRecords
+                canAddBasedOnMaxResults = allRecordsForSalesEvidenceFilter ? true : canAddBasedOnMaxResults && salesEvidenceFilter(searchParams, property)
+            }
+
+            if (canAddBasedOnMaxResults) {
+                const propertyTypeMatch = !isPropertyTypeFilterInUse && isUnderPropertyTypeLimit && matchForPropertyTypes(property, desiredPropertyTypes)
+                const propertyGroupMatch = !isPropertyGroupFilterInUse && isUnderPropertyGroupLimit && matchForPropertyGroups(property, desiredPropertyGroups)
+                const canAddBasedOnFilters = propertyGroupMatch || propertyTypeMatch
 
-                const isDupeId = uniqueSearchRecords.includes(property.id)
-                if (!isDupeId) {
-                    uniqueSearchRecords.push(property.id)
+                const ownerData = getOwnerData(property)
+                const isManaged = (property.Managed === desiredManaged) || desiredManaged === 'All'
+                const shouldAddProperty = isManaged && canAddBasedOnFilters
+
+                if (shouldAddProperty) {
+                    if (ownerData.length > 0) {
+                        property.owner_details = ownerData
+                    }
+                    if (propertyTypeMatch) {
+                        matchTallies.propertyType += 1
+                    }
+                    if (propertyGroupMatch) {
```

Comment:
Can you double check this logic? I thought it was only meant to count a property group match if it was not a property type match? Maybe things changed though.

Reply:
I put an image for the requirements of the base filter in the Zoho chat.